### PR TITLE
Create session refactor zip

### PR DESCRIPTION
**Pull request by jabrake**
This will probably still fail the CircleCI check, not sure why those errors are happening in CircleCI with Jest that I can't reproduce locally.

- Refactored `CreateSession` to use browser geolocation API to get user's latitude + longitude
- Added zip code input field to be used with Domenu API
- Fixed test errors, increased coverage to 88%

### CODE REVIEW

File: client/src/components/CreateSession.jsx [hunk 1]
```diff
@@ -12,12 +12,16 @@ class CreateSession extends React.Component {
     this.state = {
       searchQuery: '',
```
Comment (dworthm) [inline client/src/components/CreateSession.jsx:13]: I'm getting an error in the console about the input form not always being a controlled component. It looks like you set it up as one and it doesn't seem to affect anything on the page, so I think it's a minor bug.

**Review by dworthm — APPROVED** [review]

File: client/src/components/CreateSession.jsx [hunk 2]
```diff
@@ -41,10 +45,42 @@ class CreateSession extends React.Component {
...
+      const hostGeo = {
+        lat: pos.coords.latitude,
+        lon: pos.coords.longitude
+      };
+      this.props.setTopLevelState('hostGeo', hostGeo);
+      this.setState({ geolocationProcessComplete: true });
+    } catch (err) {
+      console.error(err);
+      this.setState({ geolocationProcessComplete: true });
+    }
```
Comment (milofultz) [inline client/src/components/CreateSession.jsx:83]: nit, you could add this line to a `finally` block to make it more DRY

```suggestion
      this.props.setTopLevelState('hostGeo', hostGeo);
    } catch (err) {
      console.error(err);
    } finally {
      this.setState({ geolocationProcessComplete: true });
    }
```

File: client/src/components/CreateSession.jsx [hunk 3]
```diff
@@ -124,31 +172,60 @@ class CreateSession extends React.Component {
...
+                        name="searchQuery"
+                        value={this.state.searchQuery}
+                        onChange={this.handleChange}
+                        data-testid="restaurant-search-input"
+                      />
+
+                      <label htmlFor="zipCode">Zip Code:</label>
+                      {this.props.hostGeo
+                        ? <input
+                            id="zip-code"
```
Comment (milofultz) [inline client/src/components/CreateSession.jsx:194]: The `id` has to match the label's `htmlFor`: 

> To associate the <label> with an <input> element, you need to give the <input> an id attribute. The <label> then needs a for attribute whose value is the same as the input's id.

<https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label>

File: tests/setupTests.js [hunk 4]
```diff
@@ -1,6 +1,11 @@
 const mockGeolocation = {
-  getCurrentPosition: jest.fn(),
-  watchPosition: jest.fn()
+  getCurrentPosition: jest.fn()
```
Comment (milofultz) [inline tests/setupTests.js:2]: 🔥

**Review by milofultz — CHANGES_REQUESTED** [review]
One thing on the input's id attr for the label.

File: client/src/components/CreateSession.jsx (hunk 1 above)
Comment (jabrake) [reply client/src/components/CreateSession.jsx:13]: Strange! I'm not able to reproduce locally - how did you see this? Did you pull down this branch and see how it worked on your computer? I'm seeing a console warning about input elements requiring autocomplete attributes but it looks like it's on Tom's login screen

File: client/src/components/CreateSession.jsx (hunk 2 above)
Comment (jabrake) [reply client/src/components/CreateSession.jsx:83]: Ah great catch! Updated

File: client/src/components/CreateSession.jsx (hunk 3 above)
Comment (jabrake) [reply client/src/components/CreateSession.jsx:194]: Another good catch, updated all my input id's now

**Pull request merged**